tensorflow/java/src/test/java/org/tensorflow/TensorTest.java

```diff
@@ -24,9 +24,262 @@
+        assertArrayEquals(vector, t.copyTo(actual), EPSILON);
+      }
+    }
+
+    // validate creating a tensor using a typed buffer
+    {
+      try(Tensor t = Tensor.create(DataType.INT32, scalar_shape, IntBuffer.wrap(new int[] { scalar }))) {
+        assertEquals(scalar, t.intValue());
+      }
+      try(Tensor t = Tensor.create(DataType.DOUBLE, vector_shape, DoubleBuffer.wrap(vector))) {
+        double[] actual = new double[3];
+        assertArrayEquals(vector, t.copyTo(actual), EPSILON);
+      }
+      try(Tensor t = Tensor.create(DataType.FLOAT, vector_shape, FloatBuffer.wrap(vector_f))) {
+        float[] actual = new float[3];
+        assertArrayEquals(vector_f, t.copyTo(actual), EPSILON_F);
+      }
+      int[] matrix_ = (int[]) TestUtil.flatten(matrix, Integer.TYPE);
+      try(Tensor t = Tensor.create(DataType.INT32, matrix_shape, IntBuffer.wrap(matrix_))) {
+        int[][] actual = new int[2][3];
+        assertArrayEquals(matrix, t.copyTo(actual));
+      }
+      long[] threeD_ = (long[]) TestUtil.flatten(threeD, Long.TYPE);
+      try(Tensor t = Tensor.create(DataType.INT64, threeD_shape, LongBuffer.wrap(threeD_))) {
+        long[][][] actual = new long[2][5][1];
+        assertArrayEquals(threeD, t.copyTo(actual));
+      }
+      byte[] fourD_ = TestUtil.bool2byte((boolean[]) TestUtil.flatten(fourD, Boolean.TYPE));
+      try(Tensor t = Tensor.create(DataType.BOOL, fourD_shape, ByteBuffer.wrap(fourD_))) {
+        boolean[][][][] actual = new boolean[3][1][2][4];
+        assertArrayEquals(fourD, t.copyTo(actual));
+      }
+    }
+
+    // validate that incompatible buffers are rejected
+    {
+      try {
+        Tensor.create(DataType.FLOAT, vector_shape, DoubleBuffer.wrap(vector));
+        fail("should have failed on incompatible buffer");
+      }
+      catch(IllegalArgumentException e) {
+      }
+      try {
+        Tensor.create(DataType.DOUBLE, vector_shape, FloatBuffer.wrap(vector_f));
+        fail("should have failed on incompatible buffer");
+      }
+      catch(IllegalArgumentException e) {
+      }
+      int[] matrix_ = (int[]) TestUtil.flatten(matrix, Integer.TYPE);
+      try {
+        Tensor.create(DataType.FLOAT, matrix_shape, IntBuffer.wrap(matrix_));
+        fail("should have failed on incompatible buffer");
+      }
+      catch(IllegalArgumentException e) {
+      }
+      long[] threeD_ = (long[]) TestUtil.flatten(threeD, Long.TYPE);
+      try {
+        Tensor.create(DataType.FLOAT, threeD_shape, LongBuffer.wrap(threeD_));
+        fail("should have failed on incompatible buffer");
+      }
+      catch(IllegalArgumentException e) {
+      }
+      try {
+        Tensor.create(DataType.FLOAT, new long[] { 1 }, ShortBuffer.wrap(new short[] { 1 }));
+        fail("should have failed on incompatible buffer");
+      }
+      catch(IllegalArgumentException e) {
+      }
+    }
+  }
+
+  @Test
+  public void readData() {
+
+    Tensor tscalar = Tensor.create(scalar);
```

Comment:
Let's use the [`try-with-resources`](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) statement here (which can handle multiple resources within the `try`):

```java
try (Tensor tscalar = Tensor.create(scalar);
       Tensor tvector = Tensor.create(vector);
       ....) {
   // Body of the test
}
```

That way you don't need the `finally` clause, don't have to worry about the `finally` clause covering all the tensors and it also takes care of any errors within the `create` calls.